### PR TITLE
Rename variables in linked list source files to make them more concise and standard

### PR DESCRIPTION
**Pull request by minghui-liu**
#### Bug/feature Description
The variable and function names in our linked list implementation are not very concise and standard. This PR renames some of them so that linked list related code become more readable.

#### Solution Description
Renamed some variables and functions. Below summarizes all the changes:

* `list_link.c` and `list_link.h` renamed to `linked_list.c` and `linked_list.h`.
* `pbs_list_link` and `pbs_list_head` are two names of the same structure. Renamed both to `pbs_list_node`.
* `ll_prior` and `ll_next` renamed to `prev` and `next`. The `ll` part is unnecessary and ugly IMO. `prev` is more standard than `prior` as pretty much all textbooks use `prev` and `next`. 
* `GET_PRIOR` and `get_prior` to `GET_PREV` and `get_prev`.
* `insert_link`, `append_link` and `delete_link` to `insert_node`, `append_node` and `delete_node` respectively.
* `is_linked` does not describe what the function does. Renamed to `is_in_list`.
* `CLEAR_LINK` to `CLEAR_NODE`.
* `_LIST_LINK_H` to `_LINKED_LIST_H`.

#### Checklist:

- [x] I have joined the **[pbspro community forum](http://community.pbspro.org/)**.
- [x] My pull request contains a **single, signed** commit. See **[setting up gpg signature](https://pbspro.atlassian.net/wiki/display/DG/Signing+Your+Git+Commits)**.
- [x] My code follows the **[coding style](https://pbspro.atlassian.net/wiki/display/DG/Coding+Standards)** of this project.
- [ ] My change requires project documentation. See **[required documentation checklist](https://pbspro.atlassian.net/wiki/display/DG/Checklist+for+Developing+Features+and+Bug+Fixes)** for details.
   - [ ] I have added documentation in the **[project documentation area](https://pbspro.atlassian.net/wiki/display/PD)**.
- [ ] I have added new **PTL test(s) to my commit**. (See **[using PTL for testing](https://pbspro.atlassian.net/wiki/display/DG/Using+PTL+for+Testing)**) *(or)*
   - [x] I have added  **manual test(s) to this pull request and explained why PTL is not appropriate** for this case.
- [ ] All new and existing automated tests have passed. (See **[running automated PTL tests](https://pbspro.atlassian.net/wiki/display/DG/PTL+Quick+Start+Guide)**).
- [ ] I have attached **test logs to this pull request** as evidence of testing/verification.

__***For further information please visit the [Developer Guide Home](https://pbspro.atlassian.net/wiki/display/DG/Developer+Guide+Home).***__

### CODE REVIEW

**Comment by minghui-liu** [comment]
I don't think new PTL test is needed for this PR. I compiled and started PBS without any problems. The build output is attached.
[build_output.txt](https://github.com/PBSPro/pbspro/files/1657670/build_output.txt)

**Comment by hirenvadalia** [comment]
@minghui-liu You need to update windows project files for new file names...

**Comment by mike0042** [comment]
Community discussion posted here: http://community.pbspro.org/t/pr-531-renaming-of-files-functions-and-macros/847

File: src/include/attribute.h [hunk 1]
```diff
@@ -162,7 +162,7 @@ union attr_val {	      /* the attribute value	*/
 	char 		     *at_str;	/* char string  */
 	struct array_strings *at_arst;	/* array of strings */
 	struct size_value     at_size;	/* size value */
-	pbs_list_head	      at_list;	/* list of resources,  ... */
+	pbs_list_node	      at_list;	/* list of resources,  ... */
```
Comment (subhasisb) [inline src/include/attribute.h]: I went through the change and I do not like the renaming of pbs_list_head to pbs_list_node. In current name, i.e., pbs_list_head actually is more readable, and tells me that we are passing a pointer to the head of the linked list, and not some random node in the linked list. Of course, the variable itself is called phead (in most places) signifying that it is indeed the head of the linked list, but having the type itself point that out is actually more readable (ie, if you are looking at prototypes which do not include the parameter name).

In fact the current code defines two types to make things more readable…
typedef struct pbs_list_link {
struct pbs_list_link *ll_prior;
struct pbs_list_link *ll_next;
void	*ll_struct;
} pbs_list_link;
typedef pbs_list_link pbs_list_head;

File: src/include/attribute.h [hunk 2]
```diff
@@ -189,7 +189,7 @@ typedef struct attribute attribute;
 struct attribute_def {
 	char	*at_name;
 	int	(*at_decode)(attribute *patr, char *name, char *rn, char *val);
-	int	(*at_encode)(attribute *pattr, pbs_list_head *phead,
+	int	(*at_encode)(attribute *pattr, pbs_list_node *phead,
```
Comment (subhasisb) [inline src/include/attribute.h]: I went through the change and I do not like the renaming of pbs_list_head to pbs_list_node. In current name, i.e., pbs_list_head actually is more readable, and tells me that we are passing a pointer to the head of the linked list, and not some random node in the linked list. Of course, the variable itself is called phead (in most places) signifying that it is indeed the head of the linked list, but having the type itself point that out is actually more readable (ie, if you are looking at prototypes which do not include the parameter name).

In fact the current code defines two types to make things more readable…
typedef struct pbs_list_link {
struct pbs_list_link *ll_prior;
struct pbs_list_link *ll_next;
void	*ll_struct;
} pbs_list_link;
typedef pbs_list_link pbs_list_head;

Comment (minghui-liu) [reply src/include/attribute.h]: I agree that knowing the head node in a list is a good thing. However I believe using variable name to indicate that is a better option. Having two type names for the same structure can cause confusion to new developers.

**Comment by mike0042** [comment]
Per discussion with @minghui-liu, this PR has been closed. It was agreed that linked list variable naming is an idiosyncrasy of PBS Pro that has become familiar to most developers working with the code.

**Pull request closed without merging**